dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/UserControllerTest.java

```diff
@@ -117,6 +128,41 @@ void setUp() {
     assertEquals("peter@pan.net", user.getEmail());
   }
 
+  @Test
+  void updateRolesShouldInvalidateUserSessions() {
+    CurrentUserDetails sessionPrincipal = userService.createUserDetails(superUser);
+    sessionRegistry.registerNewSession("session1", sessionPrincipal);
+
+    UserRole roleB = createUserRole("ROLE_B", "ALL");
+    userService.addUserRole(roleB);
+
+    String roleBID = userService.getUserRoleByName("ROLE_B").getUid();
+
+    PATCH(
+            "/users/" + superUser.getUid(),
+            "[{'op':'add','path':'/userRoles','value':[{'id':'" + roleBID + "'}]}]")
+        .content(HttpStatus.OK);
+
+    List<SessionInformation> allSessionsA = sessionRegistry.getAllSessions(sessionPrincipal, false);
+    assertTrue(allSessionsA.isEmpty());
+
+    sessionRegistry.registerNewSession("session2", sessionPrincipal);
+
+    PATCH(
+            "/userRoles/" + roleBID,
+            "["
+                + " {"
+                + "   'op': 'add',"
+                + "   'path': '/authorities',"
+                + "   'value': ['NONE']"
+                + " }"
+                + "]")
+        .content(HttpStatus.OK);
+
+    List<SessionInformation> allSessionsB = sessionRegistry.getAllSessions(sessionPrincipal, false);
+    assertTrue(allSessionsB.isEmpty());
```

Comment:
ideally there should be an assertion checking that it's not empty before the change is made (in case there is something wrong with the test setup)